backends/cuda/ceed-cuda-operator.c

```diff
@@ -78,12 +82,41 @@ static int CeedOperatorSetupFields_Cuda(CeedQFunction qf, CeedOperator op,
     CeedEvalMode emode;
     ierr = CeedQFunctionFieldGetEvalMode(qffields[i], &emode); CeedChk(ierr);
 
+    is_strided = false;
+    skip_restrict = false;
     if (emode != CEED_EVAL_WEIGHT) {
       ierr = CeedOperatorFieldGetElemRestriction(opfields[i], &Erestrict);
       CeedChk(ierr);
-      ierr = CeedElemRestrictionCreateVector(Erestrict, NULL,
-                                             &evecs[i + starte]);
-      CeedChk(ierr);
+
+      // Check whether this field can skip the element restriction:
+      // must be passive input, with emode NONE, and have a strided restriction with
+      // CEED_STRIDES_BACKEND.
+
+      // First, check whether the field is input or output:
+      if (!inOrOut) {
+        // Check for passive input:
+        ierr = CeedOperatorFieldGetVector(opfields[i], &field_vec); CeedChk(ierr);
+        if (field_vec != CEED_VECTOR_ACTIVE) {
+          // Check emode
+          if (emode == CEED_EVAL_NONE) {
+            // Check for strided restriction
+            ierr = CeedElemRestrictionGetStridedStatus(Erestrict, &is_strided);
+            if (is_strided) {
+              // Check if vector is already in preferred backend ordering
+              ierr = CeedElemRestrictionGetBackendStridesStatus(Erestrict, &skip_restrict);
```

Comment:
`CeedChk(ierr);`

Reply:
These were added in 8a0ea318b2c4645d3b032728c961f814ba2c3fdd